packages/contract-call-decoder/src/lib/ContractCallDecoder.ts

```diff
@@ -0,0 +1,210 @@
+import radspec from '@blossom-labs/rosette-radspec';
+import { Transaction as TransactionRosette } from '@blossom-labs/rosette-radspec/dist/declarations/src/types/web3';
+import { decode as decodeBytecode } from '@ethereum-sourcify/bytecode-utils';
+import {
+  Fragment,
+  FunctionFragment,
+  Interface,
+  JsonFragment,
+} from '@ethersproject/abi';
+import { Provider } from '@ethersproject/providers';
+import { EthereumProvider } from 'ethereum-provider';
+
+import { extractCustomFields, getValueFromDecodedFunctionData } from './utils';
+
+type Transaction = TransactionRosette & {
+  readonly chainId?: number;
+};
+
+require('isomorphic-fetch');
+
+export enum MetadataSources {
+  Sourcify,
+  BytecodeMetadata,
+}
+
+type GetMetadataOptions = {
+  readonly source?: MetadataSources;
+  readonly chainId?: number;
+  readonly address?: string;
```

Comment:
it's not mandatory because the field can be passed also in the tx object
https://github.com/ethereum/sourcify/blob/35ba02efc4c152fc550787ad515f4a6659ef2edb/packages/contract-call-decoder/src/lib/ContractCallDecoder.ts#L136